rpcs3/Emu/Cell/Modules/cellSysutil.cpp

```diff
@@ -580,16 +580,18 @@ error_code cellSysutilCheckCallback(ppu_thread& ppu)
 
 	auto& cbm = g_fxo->get<sysutil_cb_manager>();
 
-	bool read = false;
-
 	for (auto&& func : cbm.registered.pop_all())
 	{
 		if (func.call_active && !*func.call_active)
 		{
 			continue;
 		}
 
-		read = true;
+		// Increase read counter before we call the callback.
+		// We use this counter to check if the game reacts to a command during game termination and calls sys_process_exit.
+		// We would not realize that the game reacted in time and terminate it prematurely if we increased
+		// the counter after we called the callback and the callback did some time-consuming work.
+		cbm.read_counter++;
```

Comment:
save the object returned by  cbm.registered.pop_all(), increase the counter for each callback in a loop.
then add another loop that calls the callbacks.